BSP/Simulator/Src/BSP_ADC.c

```diff
@@ -0,0 +1,100 @@
+/*
+C file that reads the raw ADC data from the CSV file, 
+converts it back to a distance pressed (percentage for now), and stores both.
+We should be able to request both the raw ADC data,
+the distance (percentage for now) and millivolts from this library.
+ */
+
+//NOTE: ADC precision is 12 bits (4096), range 3.3V
+//I'm assuming 3.3V corresponds to a 100% pressed pedal
+
+#include "BSP_ADC.h"
+
+#define FILE_NAME DATA_PATH(PEDALS_CSV)
+
+/**
+ * @brief   Confirms that the CSV file
+ *          has been created and prints and 
+ *          error message if not
+ * @param   None
+ * @return  None
+ */ 
+void BSP_ADC_Init(void) {
+    // Checking if file exists
+   if(access(FILE_NAME, F_OK) != 0){
+        // File doesn't exist if true
+        perror(PEDALS_CSV);
+        exit(EXIT_FAILURE);
+    }
+    // There really isn't much we have to do in the simulator
+}
+
+/**
+ * @brief   Helper function which returns a 
+ *          milivoltage based on the raw ADC value
+ *          compared to the 3.3 volt range of the 
+ *          pedals' potentiometers
+ * @param   raw ADC value
+ * @return  milivolts 
+ */ 
+int16_t convert_ADC_to_Voltage(int16_t ADCvalue){
+    return (((float)ADCvalue / 4096) * 3300);
+}
+
+/**
+ * @brief   Helper function that reads all of the ADC values 
+ *          from a file with raw values provided 
+ *          by the ADC and stores all values in local 
+ *          array of max capacity 10
+ * @param   none
+ * @return  rawADCValues, pointer to array that contains the ADC values in the array
+ */ 
+void read_ADC_Values(int16_t *rawADCValues) {
+    //opening file in read mode
+    FILE *filePointer = fopen(FILE_NAME, "r");
+    //TODO: Correct the following function, read all values till the end of file
```

Comment:
Has this TODO been done?

Reply:
I don't think so. To use `fscanf()` and read until end of file you should have a while loop like this:

```
while(fscanf(filePointer, "%d,", &(rawADCValues[i])) == 1) {
    // Do stuff
}
```
This will break when the scanner expects 1 conversion but doesn't see anything to convert